src/tests/ftest/util/file_count_test_base.py

```diff
@@ -15,77 +17,101 @@ class FileCountTestBase(IorTestBase, MdtestBase):
             rd_fac = 'rd_fac:{}'.format(str(redundancy_factor))
             properties = (",").join(filter(None, [properties, rd_fac]))
             container.properties.update(properties)
         container.create()
 
         return container
 
+    def get_diroclass(self, rd_fac):
+        """
+        Args
+            rd_fac (int): redundancy factor
+
+        Returns:
+            str: value for dir_oclass
+        """
+
+        if rd_fac >= 2:
+            dir_oclass = "RP_3GX"
+        elif rd_fac == 1:
+            dir_oclass = "RP_2GX"
+        else:
+            dir_oclass = "SX"
+
+        return dir_oclass
+
     def run_file_count(self):
         """Run the file count test."""
         saved_containers = []
         results = []
+        dir_oclass = None
         apis = self.params.get("api", "/run/largefilecount/*")
-        object_class = self.params.get("object_class", '/run/largefilecount/*')
         hdf5_plugin_path = self.params.get("plugin_path", '/run/hdf5_vol/*')
         mount_dir = self.params.get("mount_dir", "/run/dfuse/*")
         ior_np = self.params.get("np", '/run/ior/client_processes/*', 1)
         ior_ppn = self.params.get("ppn", '/run/ior/client_processes/*', None)
         mdtest_np = self.params.get("np", '/run/mdtest/client_processes/*', 1)
         mdtest_ppn = self.params.get("ppn", '/run/mdtest/client_processes/*', None)
+        intercept = os.path.join(self.prefix, 'lib64', 'libpil4dfs.so')
+        ior_oclass = self.params.get("ior_oclass", '/run/largefilecount/object_class/*')
+        mdtest_oclass = self.params.get("mdtest_oclass", '/run/largefilecount/object_class/*')
+
         # create pool
         self.add_pool(connect=False)
 
-        for oclass in object_class:
-            self.ior_cmd.dfs_oclass.update(oclass)
-            self.mdtest_cmd.dfs_oclass.update(oclass)
-            self.ior_cmd.dfs_dir_oclass.update(oclass)
-            # oclass_dir can not be EC must be RP based on rd_fac
-            rd_fac = extract_redundancy_factor(oclass)
-            if rd_fac >= 2:
-                self.mdtest_cmd.dfs_dir_oclass.update("RP_3G1")
-            elif rd_fac == 1:
-                self.mdtest_cmd.dfs_dir_oclass.update("RP_2G1")
-            else:
-                self.mdtest_cmd.dfs_dir_oclass.update("SX")
-            for api in apis:
-                self.ior_cmd.api.update(api)
-                self.mdtest_cmd.api.update(api)
-                # update test_dir for mdtest if api is DFS
-                if api == "DFS":
-                    self.mdtest_cmd.test_dir.update("/")
-                # run mdtest
-                if self.mdtest_cmd.api.value in ['DFS', 'POSIX']:
+        for api in apis:
+            self.ior_cmd.api.update(api)
+            self.mdtest_cmd.api.update(api)
+            if api == "DFS":
+                self.mdtest_cmd.test_dir.update("/")
+            if self.mdtest_cmd.api.value in ['DFS', 'POSIX']:
+                for oclass in mdtest_oclass:
```

Comment:
Maybe it would be better if the config did this?
```
    mdtest_oclass:
      # file, dir
      - [S1, SX]
      - [EC_2P1G1, RP_2GX]
```
And then the code could do
```
for file_oclass, dir_oclass in mdtest_oclass:
```

Which would 
1. Allow the dir oclass to be configurable without modifying the code
2. Make it easier to understand what is being ran when looking at the config. Right now you have to dig into the code because it's hardcoded

Reply:
As we discussed, let's ignore this for now since you will be OOO soon